src/ripple/app/tx/impl/SetTrust.cpp

```diff
@@ -128,21 +128,42 @@ SetTrust::preclaim(PreclaimContext const& ctx)
         }
     }
 
+    auto const sleDst = ctx.view.read(keylet::account(uDstAccountID));
+
+    if (!sleDst)
+        return tecNO_DST;
+
+    auto const dstFlags = sleDst->getFlags();
+
     // If the destination has opted to disallow incoming trustlines
     // then honour that flag
     if (ctx.view.rules().enabled(featureDisallowIncoming))
     {
```

Comment:
Should it return a different error instead of `tecNO_DST` in the second case? If we get there then `featureDisallowIncoming` is not enabled.

Reply:
No, I think `tecNO_DST` makes the most sense, since there is no destination account to check for the flag. It's just that we have to check, and it's unfortunate that we have to check twice, but them's the breaks.

I suppose we could structure the check slightly differently, but then we'd be checking the rules twice:
```
// Do not reference sleDst without an amendment. Add the amendment to the if statement below so it can be checked.
auto const sleDst = ctx.view.read(keylet::account(uDstAccountID));
if (ctx.view.rules().enabled(featureDisallowIncoming) || ammEnabled(ctx.view.rules()))
{
    if (!sleDst)
        return tecNO_DST;
}
```